### PR TITLE
Ruby skeleton

### PR DESCRIPTION
**Pull request by mapanett**

### CODE REVIEW

File: fullstack/ruby/price_calculator/lib/cart/managers/cart_manager.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CartManager
+  def initialize(base_price_manager:, json_file:)
+    @base_price_manager = base_price_manager
+    @data = JSON.parse(File.read(json_file)).map { |cart_item| CartItem.from_hash(cart_item) }
+  end
+
+  def calculate_total_prices
+    raise NotImplementedError
+  end
+end
```
Comment (cosmini) [inline fullstack/ruby/price_calculator/lib/cart/managers/cart_manager.rb]:
```suggestion
end

```

EOL is inconsistent.

**Comment by cosmini** [comment]
I thought it was clear, but EOL is inconsistent across multiple files not just the one suggested.

Or we could leave those around intentionally 😁 👃

File: fullstack/ruby/price_calculator/lib/cart/models/cart_item.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class CartItem < Struct.new(:product_type, :options, :artist_markup, :quantity, :base_price)
+  def calculate_price
+    # raise NotImplementedError
+  end
+
+  def self.from_hash(hash)
+    ProductBasePrice.new(
```
Comment (tklarryonline) [inline fullstack/ruby/price_calculator/lib/cart/models/cart_item.rb]: Is this intentional?

File: fullstack/ruby/price_calculator/test/cart/managers/cart_manager_test.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+require_relative '../../../lib/cart/managers/cart_manager'
+
+class CartManagerTest < Minitest::Test
+  Dir.glob(File.join(DATA_DIR, 'cart-*.json')).each do |test_file|
+    match = /.*cart-(?<amount>\d+)\.json/.match(test_file)
+    next if match.nil?
+    expected_amount = match[:amount].to_i
+
+    define_method("test_lookup_price_and_calculate_total#{expected_amount}") do
```
Comment (tklarryonline) [inline fullstack/ruby/price_calculator/test/cart/managers/cart_manager_test.rb:12]: 🎉

**Review by tklarryonline — APPROVED** [review]
Awesome! Thanks! We can chat about the formatting but to me this looks great

File: fullstack/ruby/price_calculator/lib/cart/models/cart_item.rb (hunk 2 above)
Comment (mapanett) [reply fullstack/ruby/price_calculator/lib/cart/models/cart_item.rb]: I should say yes I intended to do that but it was a copy paste mistake.

**Pull request merged**